include/taco/lower/mode_format_impl.h

```diff
@@ -66,13 +66,13 @@ class ModeFormatImpl {
   ModeFormatImpl(std::string name, bool isFull, bool isOrdered, bool isUnique, 
                  bool isBranchless, bool isCompact, bool hasCoordValIter, 
                  bool hasCoordPosIter, bool hasLocate, bool hasInsert, 
-                 bool hasAppend);
+                 bool hasAppend, bool hasFixedSize = false, int size = 0);
```

Comment:
Maybe the constructor should just take a single argument of type `Dimension` (as defined in `type.h`) instead of `hasFixedSize` and `size`? It seems odd to have to pass these two arguments separately considering they're not really independent (i.e., if one is to specify a specific size, then the intention is presumably to make the mode format fixed size).